web/src/components/vis/VolcanoPlot.vue

```diff
@@ -23,106 +46,160 @@ export default {
-        .attr('opacity', (d) => (Math.abs(d.x) >= this.minFoldChange && d.y >= this.minLogP ? 1 : 0.5))
-        .attr('cx', (d) => this.scaleX(d.x))
-        .attr('cy', (d) => this.scaleY(d.y))
+        // @ts-ignore
+        .attr('r', (d) => (Math.abs(d.x) >= props.minFoldChange && d.y >= props.minLogP ? radius * 2 : radius))
+        // @ts-ignore
+        .attr('opacity', (d) => (Math.abs(d.x) >= props.minFoldChange && d.y >= props.minLogP ? 1 : 0.5))
+        // @ts-ignore
+        .attr('cx', (d) => _scaleX(d.x))
+        // @ts-ignore
+        .attr('cy', (d) => _scaleY(d.y))
+        // @ts-ignore
         .style('fill', (d) => d.color)
         .select('title')
+        // @ts-ignore
         .text((d) => `${d.name}: ${d.log2FoldChange} x ${d.pValue}`);
       svg.select('.plot').selectAll('line.x-threshold')
         .data([-1, 1])
         .join('line')
         .attr('class', 'x-threshold')
         .style('stroke', 'black')
         .style('stroke-width', 0.5)
-        .attr('x1', (d) => this.scaleX(d * this.minFoldChange))
+        // @ts-ignore
+        .attr('x1', (d) => _scaleX(d * props.minFoldChange))
         .attr('y1', 0)
-        .attr('x2', (d) => this.scaleX(d * this.minFoldChange))
-        .attr('y2', this.height - this.margin.top - this.margin.bottom);
+        // @ts-ignore
+        .attr('x2', (d) => _scaleX(d * props.minFoldChange))
+        // @ts-ignore
+        .attr('y2', height.value - margin.top - margin.bottom);
+
       svg.select('.plot').selectAll('line.y-threshold')
         .data([1])
         .join('line')
         .attr('class', 'y-threshold')
         .style('stroke', 'black')
         .style('stroke-width', 0.5)
         .attr('x1', 0)
-        .attr('y1', () => this.scaleY(this.minLogP))
-        .attr('x2', this.width - this.margin.left - this.margin.right)
-        .attr('y2', () => this.scaleY(this.minLogP));
-    },
+        // @ts-ignore
+        .attr('y1', () => _scaleY(props.minLogP))
+        // @ts-ignore
+        .attr('x2', width.value - margin.left - margin.right)
+        // @ts-ignore
+        .attr('y2', () => _scaleY(props.minLogP));
+    });
+
+    return {
+      width,
+      height,
+      margin,
+      dwidth,
+      dheight,
+      xlabel,
+      ylabel,
+      mainRef,
+      onResize,
+    };
   },
-};
+});
+
 </script>
-<template lang="pug">
-.main(v-resize:throttle="onResize")
-  svg(ref="svg", :width="width", :height="height", xmlns="http://www.w3.org/2000/svg")
-    g.master
-      g.axes
-      g.plot
-    text.x.label(:transform="`translate(${margin.left + dwidth / 2},${height - 10})`")
-      | {{xlabel}}
-    text.y.label(:transform="`translate(${10},${margin.top + dheight / 2})rotate(-90)`")
-      | {{ylabel}}
+<template>
+  <div
+    id="mainRef"
+    ref="mainRef"
+    v-resize:throttle="onResize"
+    class="main"
+  >
+    <svg
+      ref="svg"
```

Comment:
This ref seems unused to me.

Reply:
Addressed in https://github.com/girder/viime/pull/602/commits/8b890c887770c9a09fcaec80018901a3f408ced3